src/System.ObjectModel/tests/ObservableCollection/ObservableCollection_MethodsTest.cs

```diff
@@ -0,0 +1,843 @@
+
+            // removing null
+            anArray = new string[] { "one", "two", "three", "four" };
+            col = new ObservableCollection<string>(anArray);
+            helper = new CollectionAndPropertyChangedTester();
+            helper.RemoveItemTest(col, -1, null, false, hasDuplicates: false);
+
+            // trying to remove item in collection that has duplicates.
+            anArray = new string[] { "one", "three", "two", "three", "four" };
+            col = new ObservableCollection<string>(anArray);
+            helper = new CollectionAndPropertyChangedTester();
+            helper.RemoveItemTest(col, 1, "three", true, hasDuplicates: true);
+            // want to ensure that there is one "three" left in collection and not both were removed.
+            int occurancesThree = 0;
+            foreach (var item in col)
+            {
+                if (item.Equals("three"))
+                    occurancesThree++;
+            }
+            Assert.Equal(1, occurancesThree);
+        }
+
+        /// <summary>
+        /// Tests that a collection can be cleared.
+        /// </summary>
+        [Fact]
+        public static void ClearTest()
+        {
+            string[] anArray = { "one", "two", "three", "four" };
+            ObservableCollection<string> col = new ObservableCollection<string>(anArray);
+
+            col.Clear();
+            Assert.Equal(0, col.Count);
+            foreach (var item in col)
+            {
+                Assert.True(false, "Should not be able to iterate through an empty collection.");
+            }
+
+            try
+            {
+                string g = col[1];
+                Assert.True(false, "Should not have been able to get an item from an empty collection. Guid g: " + g);
+            }
+            catch (ArgumentOutOfRangeException)
+            {
+            }
+
+            //tests that the collectionChanged events are fired.
+            CollectionAndPropertyChangedTester helper = new CollectionAndPropertyChangedTester();
+            col = new ObservableCollection<string>(anArray);
+            helper.ClearTest(col);
+        }
+
+        /// <summary>
+        /// Tests that we can remove items at a specific index, at the middle beginning and end.
+        /// </summary>
+        [Fact]
+        public static void RemoveAtTest()
+        {
+            Guid[] anArray = { Guid.NewGuid(), Guid.NewGuid(), Guid.NewGuid() };
+            ObservableCollection<Guid> col0 = new ObservableCollection<Guid>((IEnumerable<Guid>)anArray);
+            ObservableCollection<Guid> col1 = new ObservableCollection<Guid>((IEnumerable<Guid>)anArray);
+            ObservableCollection<Guid> col2 = new ObservableCollection<Guid>((IEnumerable<Guid>)anArray);
+
+            col0.RemoveAt(0);
+            string collectionString = "";
+            foreach (var item in col1)
+                collectionString += item + ", ";
```

Comment:
What is collectionString being used for here?  It looks like we're building it up but then never looking at its value.  Are we somehow testing the collection's enumeration?  And if so, are we supposed to be enumerating col0 rather than col1?